### DISCUSSION TITLE
Should we consider adding "report-uri" as a Content-Security-Policy header setting

### DISCUSSION THREAD

**Discussion by yaleman** [kanidm/kanidm#850] [Ideas]
It's one of those security tweaks that could be handy, yes I know it's yet another knob to tweak but with IdP's, seeeeecurity.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-uri

tl;dr should we have a configurable setting for the `report-uri` CSP header on kanidmd, defaulting to `'none'` if unset.

**Comment by Firstyear** [comment]
It says it's deprecated on MDN ... ?

**Reply by yaleman** [reply]
Yeah, but it's  the only thing currently supported by most browsers and [report-to](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-to ) is the new/updated one with a similar format - Mozilla notes it as deprecated but Firefox doesn't support the new one 😕

Maybe the setting could be better articulated as a CSP report URL, which can get put into both headers.

**Reply by Firstyear** [reply]
Yeah, could be worth supporting both then :)

**Comment by Firstyear** [comment]
Should this be a dev issue?

**Reply by yaleman** [reply]
Yepyep, I figured a "should we" discussion was better before a "we should" dev issue 😄

**Reply by yaleman** [reply]
Raised as #1169 (nice?)

**Reply by Firstyear** [reply]
nice.
